addons/stock_account/migrations/13.0.1.1/post-migration.py

```diff
@@ -0,0 +1,333 @@
+            ORDER BY company_id, product_id, datetime
+            """, (company.id, product.id, move.date, last_svl.create_date))
+            pph_data = env.cr.fetchall()
+            price_at_date = last_svl.unit_cost
+            for pph in pph_data:
+                diff = pph[4] - price_at_date
+                value = company.currency_id.round(quantity_svl * diff)
+                if not company.currency_id.is_zero(value):
+                    vals = {
+                        'company_id': company.id,
+                        'product_id': product.id,
+                        'description': _('Product value manually modified (from %s to %s)') % (price_at_date, pph[4]),
+                        'value': value,
+                        'quantity': 0,
+                        'old_product_price_history_id': pph[0],
+                        'create_date': pph[3],
+                        'create_uid': pph[5] or 1,
+                        'write_uid': pph[6] or 1,
+                        'write_date': pph[7],
+                    }
+                    _create_stock_valuation_layer(env, [vals])
+                    std_price_update[company.id, product.id] = pph[4]
+
+
+def _get_price_unit(move, std_price_update):
+    """ Returns the unit price to value this stock move """
+    move.ensure_one()
+    price_unit = move.price_unit
+    # If the move is a return, use the original move's price unit.
+    returned_svl = move.env['stock.valuation.layer'].search([
+        ('stock_move_id', '=', move.origin_returned_move_id.id),
+    ], order='create_date desc, id desc', limit=1)
+    if move.origin_returned_move_id and returned_svl:
+        price_unit = returned_svl.unit_cost
+    if not move.company_id.currency_id.is_zero(price_unit):
+        return price_unit
+    return std_price_update.get(move.company_id.id, move.product_id.id) or move.product_id.standard_price
+
+
+def _create_stock_valuation_layer(env, vals_list):
+    for vals in vals_list:
+        if 'account_move_id' not in vals:
+            vals['account_move_id'] = _get_related_account_move(env, vals).id or None
+        columns = vals.keys()
+        query = """
+            INSERT INTO stock_valuation_layer ({})
+            VALUES({})
+        """.format(', '.join(columns), ", ".join(["%({})s".format(col) for col in columns]))
+        openupgrade.logged_query(env.cr, query, vals)
+
+
+def _get_related_account_move(env, svl_vals):
+    """ Return Account move related to Stock Valuation Layer"""
+    domain = []
+    if svl_vals.get('stock_move_id'):
+        domain = expression.AND([domain, [('stock_move_id', '=', svl_vals['stock_move_id'])]])
+    if svl_vals.get('old_product_price_history_id'):
+        env.cr.execute("""
+            SELECT create_date
+            FROM product_price_history
+            WHERE id = %s
+        """, (svl_vals['old_product_price_history_id'],))
+        create_date = env.cr.fetchone()[0]
```

Comment:
Is it need to add `LIMIT 1` here?

Reply:
I think it doesn't matter in this case, because fetchone already limits to one. The select is not logged, so also don't care if the output is short or long. But I can add it if you want.

Reply:
OK, no problem.
I wanted just ask about it?

Reply:
There is always an index on the `id` column in Odoo, so when selecting a record by that column there is no benefit in adding a LIMIT.